src/expression.d

```diff
@@ -2831,7 +2831,22 @@ extern (C++) abstract class Expression : RootObject
             assert(type);
             if (!type.isMutable())
             {
-                error("cannot modify %s expression %s", MODtoChars(type.mod), toChars());
+                bool errFromFunc;
+                for (Dsymbol s = sc.func; s; s = s.toParent2())
+                {
+                    FuncDeclaration ff = s.isFuncDeclaration();
+                    if (!ff)
+                        break;
+                    if (!ff.type.isMutable)
```

Comment:
How can I can I determine if the expression uses `this` ?

`if (memcmp(rep, "this.".ptr, 5) == 0) ` does the the trick but this doesn't look good.